packages/e2e-tests/specs/experiments/block-directory-add.test.js

```diff
@@ -0,0 +1,169 @@
+
+	registerBlockType( '${ MOCK_BLOCK1.name }', {
+		title: 'Test Block for Block Directory',
+		icon: 'hammer',
+		category: 'common',
+		attributes: {},
+		edit: function( props ) {
+			return el( 'p', null, 'Test Copy' );
+		},
+		save: function() {
+			return null;
+		},
+	} );
+} )();`;
+
+const MOCK_EMPTY_RESPONSES = [
+	{
+		match: ( request ) => matchUrl( request.url(), SEARCH_URLS ),
+		onRequestMatch: createResponse( JSON.stringify( [] ) ),
+	},
+];
+
+const MOCK_BLOCKS_RESPONSES = [
+	{
+		// Mock response for search with the block
+		match: ( request ) => matchUrl( request.url(), SEARCH_URLS ),
+		onRequestMatch: createResponse(
+			JSON.stringify( [ MOCK_BLOCK1, MOCK_BLOCK2 ] )
+		),
+	},
+	{
+		// Mock response for install
+		match: ( request ) => matchUrl( request.url(), INSTALL_URLS ),
+		onRequestMatch: createResponse( JSON.stringify( true ) ),
+	},
+	{
+		// Mock the response for the js asset once it gets injected
+		match: ( request ) => request.url().includes( MOCK_BLOCK1.assets[ 0 ] ),
+		onRequestMatch: createResponse(
+			Buffer.from( block, 'utf8' ),
+			'application/javascript; charset=utf-8'
+		),
+	},
+];
+
+function getResponseObject( obj, contentType ) {
+	return {
+		status: 200,
+		contentType,
+		body: obj,
+	};
+}
+
+function createResponse( mockResponse, contentType = undefined ) {
+	return async ( request ) =>
+		request.respond( getResponseObject( mockResponse, contentType ) );
+}
+
+const matchUrl = ( reqUrl, urls ) => {
+	return urls.some( ( el ) => reqUrl.indexOf( el ) >= 0 );
+};
+
+describe( 'adding blocks from block directory', () => {
+	beforeEach( async () => {
+		await enableExperimentalFeatures( [ '#gutenberg-block-directory' ] );
+		await createNewPost();
+	} );
+
+	it( 'Should show an empty state when no plugin is found.', async () => {
+		// Be super weird so there won't be a matching block installed
+		const impossibleBlockName = '@#$@@Dsdsdfw2#$@';
+
+		// Return an empty list of plugins
+		await setUpResponseMocking( MOCK_EMPTY_RESPONSES );
+
+		// Search for the block via the inserter
+		await searchForBlock( impossibleBlockName );
+
+		const selectorContent = await page.evaluate(
+			() =>
+				document.querySelector( '.block-editor-inserter__block-list' )
+					.innerHTML
+		);
+		expect( selectorContent ).toContain( 'has-no-results' );
+	} );
+
+	it( 'Should be able to add (the first) block.', async () => {
+		// Setup our mocks
+		await setUpResponseMocking( MOCK_BLOCKS_RESPONSES );
+
+		// Search for the block via the inserter
+		await searchForBlock( MOCK_BLOCK1.title );
+
+		// Grab the first block in the list -> Needs to be the first one, the mock response expects it.
+		const addBtn = await page.waitForSelector(
+			'.block-directory-downloadable-blocks-list li:first-child button'
+		);
+
+		// Add the block
+		await addBtn.click();
+
+		// To be honest, not sure why this is necessary, but the getEditedPostContent returns empty without querying wp.data first.
+		await getAllBlocks();
```

Comment:
I _think_ this just needs time for the block script to load— I can get a more reliable passing rate with a fixed delay.

```suggestion
		// Delay to let block script load
		await new Promise( ( resolve ) => setTimeout( resolve, 100 ) );
```

Reply:
Oh nice. I don't know why I didn't try a timeout. I thought it was state related. I'll update to use the timeout for now. It's more communicative.